src/components/data/QueryAuth0.js

```diff
@@ -0,0 +1,56 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { fetchUserProfile, logUserIn } from 'console/state/auth/actions';
+import { getAccessToken } from 'console/state/auth/selectors';
+import { finishAuthenticationFlow } from 'console/utils/auth0';
+
+@connect(
+  (state, props) => ({
+    accessToken: getAccessToken(state),
+  }),
+  {
+    fetchUserProfile,
+    logUserIn,
+  },
+)
+@withRouter
+export default class QueryAuth0 extends React.PureComponent {
+  static propTypes = {
+    accessToken: PropTypes.string,
+    fetchUserProfile: PropTypes.func.isRequired,
+    history: PropTypes.object.isRequired,
+    logUserIn: PropTypes.func.isRequired,
+  };
+
+  async componentWillMount() {
+    let authResult;
+
+    try {
+      authResult = await finishAuthenticationFlow();
+    } catch (err) {
+      // TODO: Handle login failure
+      console.log(err);
```

Comment:
Perhaps not having the whole try-catch and just let it throw errors for now. And a "TODO" comment that is the URL to a new issue about this.